version/main.go

```diff
@@ -0,0 +1,477 @@
+		WithCommand([]string{"fetch", "--no-tags", "origin", "+refs/tags/*:refs/tags/*", "--depth=1"})
+}
+
+type VersionTag struct {
+	Component string
+	Version   string
+	Commit    string
+}
+
+// Return a list of version tags pointing to the current commit
+// A version tag is of the form [PATH/]SEMVER. Examples:
+//   - "v0.1.0"
+//   - "sdk/php/v0.1.1"
+//   - "bla/v1.1.0"
+func (v Version) VersionTags(ctx context.Context) ([]VersionTag, error) {
+	if !v.gitDirExists(ctx) {
+		return nil, nil
+	}
+	commit, err := v.Commit(ctx)
+	if err != nil {
+		return nil, err
+	}
+	tagsRaw, err := v.gitRepo().
+		Command([]string{"tag", "--points-at", "HEAD"}).
+		Stdout(ctx)
+	if err != nil {
+		return nil, err
+	}
+	var tags []VersionTag
+	for _, tag := range strings.Split(strings.Trim(tagsRaw, "\n"), "\n") {
+		if semver.IsValid(tag) {
+			tags = append(tags, VersionTag{Version: tag, Commit: commit})
+			continue
+		}
+		// Split the path by "/" and get the last part
+		parts := strings.Split(tag, "/")
+		if len(parts) < 2 {
+			continue // Ensure there are at least two path parts and the version
+		}
+		version := parts[len(parts)-1]
+		// Check if the last part is a valid semver
+		if !semver.IsValid(version) {
+			continue
+		}
+		tags = append(tags, VersionTag{
+			Version:   version,
+			Component: strings.Join(parts[:len(parts)-1], "/"),
+			Commit:    commit,
+		})
+	}
+	return tags, nil
+}
+
+// Return all tags pointing at the current commit
+func (v Version) Tags(ctx context.Context) ([]string, error) {
+	if !v.gitDirExists(ctx) {
+		return nil, nil
+	}
+	commit, err := v.Commit(ctx)
+	if err != nil {
+		return nil, err
+	}
+	tagsRaw, err := v.gitRepo().
+		Command([]string{"tag", "--points-at", commit}).
+		Stdout(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return strings.Split(strings.Trim(tagsRaw, "\n"), "\n"), nil
+}
+
+// Return the semver-compatible tag pointing to the current tag, or an empty string
+func (v Version) VersionTag(ctx context.Context) (string, error) {
+	tags, err := v.VersionTags(ctx)
+	if err != nil {
+		return "", err
+	}
+	if len(tags) == 0 {
+		return "", nil
+	}
+	for _, tag := range tags {
+		if tag.Component == "" {
+			return tag.Version, nil
+		}
+	}
+	return "", nil
+}
+
+func (v Version) gitDirExists(ctx context.Context) bool {
+	_, err := v.GitDir.Directory(".git").Entries(ctx)
+	return err == nil
+}
+
+// Return the current git commit
+func (v Version) Commit(ctx context.Context) (string, error) {
+	if !v.gitDirExists(ctx) {
+		return "", nil
+	}
+	cmd := []string{"rev-parse", "HEAD"}
+	// FIXME: contribute GitRepo.head() uptsream
+	commit, err := v.gitRepo().Command(cmd).Stdout(ctx)
+	if err != nil {
+		return "", err
+	}
+	return strings.Trim(commit, "\n"), nil
+}
+
+// Compute a stable digest of the input files, to differentiate dev builds on the same commit
+// FIXME: the digest doesn't seem very stable...
+func (v Version) InputsDigest(ctx context.Context) (string, error) {
```

Comment:
This and some of the other functions in here should be private - since they're internal implementation details.